addons/project/models/project.py

```diff
@@ -472,9 +472,11 @@ def _read_group_stage_ids(self, stages, domain, order):
 
     def _compute_attachment_ids(self):
         for task in self:
-            attachment_ids = self.env['ir.attachment'].search([('res_id', '=', task.id), ('res_model', '=', 'project.task')]).ids
-            message_attachment_ids = self.mapped('message_ids.attachment_ids').ids  # from mail_thread
-            task.attachment_ids = list(set(attachment_ids) - set(message_attachment_ids))
+            message_attachments = task.message_ids.mapped('attachment_ids')  # from mail_thread
+            attachments = self.env['ir.attachment'].search([
+                ('res_id', '=', task.id), ('res_model', '=', 'project.task'),
+                ('id', 'not in', message_attachments.ids)])
+            task.attachment_ids = attachments | message_attachments
```

Comment:
Why do you use `attachments | message_attachments`?

When you search for attachments, you exclude messages's attachments, using
`('id', 'not in', message_attachments.ids)])`

So, why you exclude them, but later include them using `|`? (union operator).

Reply:
I'm with Luis: you don't need to add again `message_attachments`. In fact, original implementation exclude them (`task.attachment_ids = list(set(attachment_ids) - set(message_attachment_ids))`)

Reply:
You are right!
It is a `-` instead of a `|`
Thanks!

Reply:
But they are already excluded in the `search` sentence AFAIK